src/main/java/io/cdap/plugin/decompress/action/DecompressActionConfig.java

```diff
@@ -134,7 +135,25 @@ public void validate(FailureCollector collector) {
         .withStacktrace(e.getStackTrace());
     }
 
-    if (!containsMacro(SOURCE_FILE_PATH) && Strings.isNullOrEmpty(sourceFilePath)) {
```

Comment:
why change this? It looks like this is only triggered when the path is not a macro and it's missing, which looks correct.

Reply:
The actual issue is below with the line `Path source = new Path(sourceFilePath);`. That fails with an NPE if macro substitution happened.

Reply:
sourceFilePath is not null after macro substitution. I still don't see what was wrong with the earlier code, were you able to make it error?

Reply:
the jira mentions version 1.0.0 of this plugin. Develop is 1.2.0-snapshot

Reply:
Yes, I was able to reproduce the error because `this.sourceFilePath` is null when I click the validate button. I tried this in 1.2.0-SNAPSHOT.

As for versioning, I can cherry-pick the version needed, but this code requires at least CDAP 6.1.2.

Reply:
containsMacro() is supposed to return true in that situation. If it's not, it's a problem in the pipeline studio service. It's not a problem with the plugin.

Reply:
though I don't know why this is being checked at all. The source path is not annotated as nullable, which means the framework will make sure the variable is never set to null unless there is a macro.

The file path also shouldn't be checked during validation or deployment time at all, it should just be checked at runtime. I think it's better to just remove this entire block of code, it's validating things it shouldn't be validating.

Reply:
But the test case tries to pass in an empty source and dest file path, so can we remove those test cases?

Reply:
Can the test be fixed if it not doing what it is supposed to do?

Reply:
@chtyim there are test cases that do nothing besides verify that an empty path will cause an error to be thrown. Can we delete those cases?

Reply:
A non-macro empty path in the configuration is an error which needs to be detected. Why it is not needed? It is different than what Albert mentioned above that if the path is provided, we shouldn't check if the path actually exist. However, if the path provided, which is not a macro, but is invalid, when it should fail.

Reply:
so you're saying we should do something like this:

``` if (!containsMacro(FIELD_NAME) && Strings.isEmpty(field)) {collector.addFailure(...);}```

So some validation is expected, but not checking the file path scheme. Is this correct?

Reply:
Something along this line. What is a valid path? Is there any restrictions? Does it has to be a valid, parsable URI?

Reply:
I believe the original code tried to validate every scheme with Java's built-in validator, but that doesn't make sense to me since it rejects `gs://` paths